### ISSUE TITLE
Quality control for JPEG 2000 images

### ISSUE THREAD

**Issue by pkorus** [imageio/imageio#453]
imageio version: 2.5.0

I'm having trouble controlling image quality for JPEG 2000 images. `imageio.help('j2c')` lists a number of parameters that should be supported e.g., `quality_mode` and `quality_layers` but the current code base does not seem to support any. The format class which ends up being used is `PillowFormat` which does not support any keyword arguments and ends up throwing an exception:

```
>> imageio.imwrite('test.j2c', image, quality_layers=[1])
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/imageio/core/functions.py", line 259, in imwrite
    writer = get_writer(uri, format, "i", **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/imageio/core/functions.py", line 187, in get_writer
    return format.get_writer(request)
  File "/usr/local/lib/python3.5/dist-packages/imageio/core/format.py", line 178, in get_writer
    return self.Writer(self, request)
  File "/usr/local/lib/python3.5/dist-packages/imageio/core/format.py", line 214, in __init__
    self._open(**self.request.kwargs.copy())
TypeError: _open() got an unexpected keyword argument 'quality_layers'
```

The actual function being called is `plugins.pillow.PillowFormat.Writer._open`:
```python
...
        def _open(self):
            Image = self.format._init_pillow()
            try:
                self._save_func = Image.SAVE[self.format.plugin_id]
            except KeyError:
                raise RuntimeError("Format %s cannot write images." % self.format.name)
            self._fp = self.request.get_file()
            self._meta = {}
            self._written = False
...
```

Are there any plans to fix this? Or does anyone know a possible workaround?

**Comment by almarklein** [comment]
This needs some explanation. Imageio uses Pillow to support many different common formats. Most of these are autogenerated and only support arguments `pilmode`, `as_gray` and `ignoregamma`. For some formats we have custom format classes to give control of more specifics, e.g. png and jpg. We don't have such a custom class for JPEG2000, yet.

One solution would be to implement such as custom class for JPEG2000. Have a look here for reference: https://github.com/imageio/imageio/blob/master/imageio/plugins/pillow.py#L369

You could do this for yourself, and add make imageio aware using `imageio.formats.add_format()`. Though if you have made such a class, it would be great if we could add it to imageio :D

**Comment by pkorus** [comment]
I've added a simple class & issued a pull request.

**Comment by FirefoxMetzger** [comment]
@almarklein There is no PR linked to this issue, but that doesn't mean that it didn't happen. Do you remember if this was resolved? If so, can we close the issue?

If not, it sounds like our new API will also take care of this, since it allows pillow-native `kwarg`s to be passed on. I think this should resolve the question, too, right? If not, I'd need some more details on what is missing so that we can see how to fit this into the new pillow plugin.

**Comment by almarklein** [comment]
Closed by #456
